src/element.rs

```diff
@@ -46,3 +46,15 @@ implement!(usize);
 
 implement!(c32, c32::new(0.0, 0.0));
 implement!(c64, c64::new(0.0, 0.0));
+
+impl<T> Element for Option<T> where T: Copy + PartialEq {
```

Comment:
It seems very application specific. In my opinion, this code doesn’t belong to this crate. I suggest defining your own version of `Option` in your code and implementing `Element` for that special `Option`. Then you can twist and tune it in whatever way that fits your application best.